rasa/shared/core/domain.py

```diff
@@ -298,9 +299,13 @@ def merge_lists_of_dicts(
         if override or self.session_config == SessionConfig.default():
             combined[SESSION_CONFIG_KEY] = domain_dict[SESSION_CONFIG_KEY]
 
-        combined[KEY_INTENTS] = merge_lists_of_dicts(
-            combined[KEY_INTENTS], domain_dict[KEY_INTENTS], override
-        )
+        duplicates: Dict[Text, List[Text]] = {}
+
+        # this code merges lists of dicts of intents
+        dict1 = {list(i.keys())[0]: i for i in combined[KEY_INTENTS]}
+        dict2 = {list(i.keys())[0]: i for i in domain_dict[KEY_INTENTS]}
+        duplicates[KEY_INTENTS] = extract_duplicates(dict1, dict2)
```

Comment:
Did we consider just raising an error/warning here so that we don't need to store the duplicates on the class?

Reply:
The idea was to show this information only when running `rasa data validate`, as it's written in the description of this issue. We can update it to show the warning anyway but in this case I don't see the point in doing this sort of check in `rasa data validate` which will mean the whole task will become different.

Reply:
Maybe @TyDunn can take a look at it and say which approach we prefer.

Reply:
If we do it the way @joejuzl proposes then the warnings/exceptions will be raised every time the domain is getting merged (e.g. when just running `rasa train` or literally any other command that uses domain), not only when doing `rasa data validate`

Reply:
@TyDunn If we do it the way @joejuzl proposes then the warnings/exceptions will be raised every time the domain is getting merged (e.g. when just running rasa train or literally any other command that uses domain), not only when doing rasa data validate

Reply:
@alwx I believe Joe is out for the rest of the year, and I am not going to be able to gain enough context atm. Can you make a judgement call here yourself?

Reply:
@TyDunn I think it makes sense to keep this one as it as and maybe create a new issue because what Joe was talking about makes the scope much bigger